doc/tutorials/BasicValidators.hs

```diff
@@ -82,16 +83,16 @@ validatePayment _ _ ctx =
     in check $ valueOf (fold values) adaSymbol adaToken >= 1
 --- BLOCK5
 -- We can serialize a 'Validator' directly to CBOR
-serializedDateValidator :: BSL.ByteString
-serializedDateValidator = serialise dateValidator
+serialisedDateValidator :: SerialisedScript
+serialisedDateValidator = serialiseScript dateValidator
 
 -- The serialized forms can be written or read using normal Haskell IO functionality.
 showSerialised :: IO ()
-showSerialised = print serializedDateValidator
+showSerialised = print serialisedDateValidator
 -- BLOCK6
 -- We can serialize 'CompiledCode' also
-serializedCompiledCode :: BS.ByteString
-serializedCompiledCode = Flat.flat $ alwaysSucceedsCompiled
+serialisedCompiledCode :: BS.ByteString
+serialisedCompiledCode = Flat.flat alwaysSucceedsCompiled
```

Comment:
This tutorial is confusing. We mention CBOR forSerialisedScript and DateValidator, but we use flat for serialisedCompiledCode and loadFromFile

We should say that the on-chain serialized format is in CBOR but the off-chain format for locally storing and executing plutus code is in flat.

Alternatively, we can cleanup the confusion between CompiledCode (Flat) and Script (CBOR). From a user's perspective, it is confusing to explain that there is CompiledCode but also on-chain Script and they both have different serialization formats; in the end they are both plutus scripts.

I propose to have the same serialization format for both off-chain and on-chain scripts (CBOR), but differentiate slightly between offline `DebugScript` and on-chain `Script`.

```
type DebugScript = Program NamedDeBruijn ... (serialized via CBOR-enveloped Flat)
type Script = Program DeBruijn (serialised via CBOR-enveloped Flat)
```

what do you think?

Reply:
I think we should use CBOR everywhere and keep flat as an implementation detail.

Reply:
Yeah nobody outside of our team should need to know anything about Flat, or what CompiledCode is. All they need to know is `serialiseScript` (I probably should name it `serialiseCompiledCode`) serialises CompiledCode into CBOR.

I do agree that the tutorial is confusing. It shouldn't use Flat. In fact I don't even know whether this tutorial is useful at all. It's a pretty old tutorial and we should decide if we should keep it or come up with a better one as part of Joseph's work of re-organizing the readthedocs site.

As to the type aliases: there's already `SerialisedScript` that wraps `Program NamedDeBruijn`. We could have a type/newtype that wraps `Program DeBruijn`, but the ledger doesn't need it, and so I'd expect it to be defined in Plutus Tools.